layout/layout.go

```diff
@@ -0,0 +1,740 @@
+	cfg, err := i.Image.ConfigFile()
+	if err != nil {
+		return "", errors.Wrapf(err, "getting config file for image at path %q", i.path)
+	}
+	if cfg == nil {
+		return "", fmt.Errorf("missing config for image at path %q", i.path)
+	}
+	labels := cfg.Config.Labels
+	return labels[key], nil
+}
+
+func (i *Image) Labels() (map[string]string, error) {
+	cfg, err := i.Image.ConfigFile()
+	if err != nil {
+		return nil, errors.Wrapf(err, "getting config file for image at path %q", i.path)
+	}
+	if cfg == nil {
+		return nil, fmt.Errorf("missing config for image at path %q", i.path)
+	}
+	return cfg.Config.Labels, nil
+}
+
+func (i *Image) Env(key string) (string, error) {
+	cfg, err := i.Image.ConfigFile()
+	if err != nil {
+		return "", errors.Wrapf(err, "getting config file for image at path %q", i.path)
+	}
+	if cfg == nil {
+		return "", fmt.Errorf("missing config for image at path %q", i.path)
+	}
+	for _, envVar := range cfg.Config.Env {
+		parts := strings.Split(envVar, "=")
+		if parts[0] == key {
+			return parts[1], nil
+		}
+	}
+	return "", nil
+}
+
+func (i *Image) WorkingDir() (string, error) {
+	cfg, err := i.Image.ConfigFile()
+	if err != nil {
+		return "", errors.Wrapf(err, "getting config file for image at path %q", i.path)
+	}
+	if cfg == nil {
+		return "", fmt.Errorf("missing config for image at path %q", i.path)
+	}
+	return cfg.Config.WorkingDir, nil
+}
+
+func (i *Image) Entrypoint() ([]string, error) {
+	cfg, err := i.Image.ConfigFile()
+	if err != nil {
+		return nil, errors.Wrapf(err, "getting config file for image at path %q", i.path)
+	}
+	if cfg == nil {
+		return nil, fmt.Errorf("missing config for image at path %q", i.path)
+	}
+	return cfg.Config.Entrypoint, nil
+}
+
+func (i *Image) OS() (string, error) {
+	cfg, err := i.Image.ConfigFile()
+	if err != nil {
+		return "", errors.Wrapf(err, "getting config file for image at path %q", i.path)
+	}
+	if cfg == nil {
+		return "", fmt.Errorf("missing config for image at path %q", i.path)
+	}
+	if cfg.OS == "" {
+		return "", fmt.Errorf("missing OS for image at path %q", i.path)
+	}
+	return cfg.OS, nil
+}
+
+func (i *Image) OSVersion() (string, error) {
+	cfg, err := i.Image.ConfigFile()
+	if err != nil {
+		return "", errors.Wrapf(err, "getting config file for image at path %q", i.path)
+	}
+	if cfg == nil {
+		return "", fmt.Errorf("missing config for image at path %q", i.path)
+	}
+	return cfg.OSVersion, nil
+}
+
+func (i *Image) Architecture() (string, error) {
+	cfg, err := i.Image.ConfigFile()
+	if err != nil {
+		return "", errors.Wrapf(err, "getting config file for image at path %q", i.path)
+	}
+	if cfg == nil {
+		return "", fmt.Errorf("missing config for image at path %q", i.path)
+	}
+	if cfg.Architecture == "" {
+		return "", fmt.Errorf("missing Architecture for image at path %q", i.path)
+	}
+	return cfg.Architecture, nil
+}
+
+func (i *Image) Name() string {
+	return i.path
+}
+
+func (i *Image) Rename(name string) {
+	i.path = name
```

Comment:
If `Rename` is called and the image is saved, does it copy/move the bits?

Reply:
- If you call the rename `before` saving, then the image is going to be saved in the new destination
- If you call the rename `after` saving and then call the `save` again, then the image is going to be saved in the new path, but the original one is not deleted or anything. Maybe it could be smarter but I do not want to complicate the implementation right now.